react/provider/providerUser.tsx

```diff
@@ -163,6 +171,32 @@ const ProviderUser: FC = (props: Props) => {
     length: ITEMS_PER_PAGE,
   })
 
+  useEffect(() => {
+    if (loadingSearchGiftCard || loadingSearchUser || loadingSearchListUser) {
+      setLoading(true)
+    }
+
+    if (
+      !loadingSearchGiftCard &&
+      !loadingSearchUser &&
+      !loadingSearchListUser
+    ) {
+      setLoading(false)
+    }
+  }, [loadingSearchGiftCard, loadingSearchListUser, loadingSearchUser])
```

Comment:
What do you think about change this `useEffet` with `useState` to a `useMemo`?
Because you can set the return of these loading properties to a constant and use it how you are using the `loading` from `useState` currently.

```suggestion
  const loading = useMemo(() => loadingSearchGiftCard || loadingSearchUser || loadingSearchListUser)
```

Maybe it works good

Reply:
I think it's an amazing idea
I will try here and if it works as well, I will implement